### PR TITLE
Restyle Countdown block

### PR DESCRIPTION
**Pull request by teenoh**
## 🌟 What is the purpose of this PR?

This PR updates the styling of countdown block
- [Figma Designs](https://www.figma.com/file/T10d2P2iNb1LdRC5a9J7vA/BP-Block-Week-Designs?node-id=49%3A9164) (internal link)

## 🔗 Related links

- [Implement Countdown block designs](https://app.asana.com/0/1202176479837585/1202255059566946) _(internal)_

## 🔍 What does this change?

- Updates the countdown block UI to match designs
- Breaks the existing app into smaller components 
- Removes range selection feature. [internal link](https://hashintel.slack.com/archives/C02LG39FJAU/p1653317467951639) for more context
- Adds a `displayTime` prop which is used to determine if time (hours, mins, seconds) should also be displayed in the countdown

## 📜 Does this require a change to the docs?

-No

## ⚠️ Known issues

- The time input makes use of the native time picker and it's not possible to override the picker styles to match our designs on chrome. We'd need to create a custom time picker instead.

Screenshot 2022-05-25 at 15 38 57

## 🐾 Next steps

- Create a custom time picker

## ❓ How to test this?

1.  Run the block locally or load into the workspace app
1.  Confirm it matches attached design and functionality is the same

## 📹 Demo

https://user-images.githubusercontent.com/16928688/170277498-f942d02c-e7bd-4139-bacb-f134bb256ac1.mov

### CODE REVIEW

File: packages/blocks/countdown/src/base.css [hunk 1]
```diff
@@ -0,0 +1,395 @@
+@import url("https://fonts.googleapis.com/css2?family=Inter:wght@300;400;500;600;700;800&display=swap");
```
[the remark this answers is not in the document]
Comment (nonparibus) [reply packages/blocks/countdown/src/base.css:1]: False positive

File: packages/blocks/countdown/src/base.css [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+/* –––––––––– Root variables –––––––––– */
+
+:root {
```
Comment (kachkaev) [inline packages/blocks/countdown/src/base.css:5]: This CSS is incompatible with un-sandboxed blocks. Not sure what to replace this CSS with if we want optional sandboxing. @CiaranMn any thoughts?

Comment (CiaranMn) [reply packages/blocks/countdown/src/base.css:5]: We spoke about this – in this case it looks like the variables are mostly un-used, so I would move any that are actually used to `styles.css`, where there are already some scoped, and delete the file.

We could additionally make `styles.css` a CSS module (renaming it to `styles.module.css` might work automatically, unsure) to ensure no clashes with other classes called `display-grid`.

Comment (teenoh) [reply packages/blocks/countdown/src/base.css:5]: I scoped the styles to the block's class name for now in [5d5eecb](https://github.com/hashintel/hash/pull/583/commits/5d5eecb73e9332c071fc32b269b9e35ee3b68eda). Couldn't find a quick fix to the bug I mentioned earlier about the imported styles object being empty. I believe that will be address in [internal task](https://app.asana.com/0/1201095311341924/1202361526477995)

Comment (CiaranMn) [reply packages/blocks/countdown/src/base.css:5]: We can just delete this then, right? It's still adding global styles that it shouldn't

**Comment by MaggieAppleton** [comment]
Made a couple small changes to the CSS to align with the figma. Also added a few refinements like animations transitions between states.

- The seconds counter was quite distracting so I removed it. If we're expecting people will use this to countdown to major dates at the scale of days, weeks, and months, it's unlikely they'll want seconds to show all the time.
- Removed the count for number of weeks and adjusted the margin on the days of the weeks so they align with the calendar month numbers
- Made the event title input hug width to its content length

There's one small bug I can't figure out how to fix with .scss. We'd be able to do it with styled-components, but don't know the work around without it.

When there's no time set on the picker, it should be a width of 100px; When there is a time set we need the picker to grow to 160px. Ideally the picker would just wrap to fit its contents, but because for some reason the text input isn't responding to either `width: min-content` or `width: fit-content`

Should look like this:

100px w/o time
image

160px w/ time
image

But I can't figure out how to pass the value of the `displayTime` prop down into the scss to conditionally apply it. For now I've just set the width to 100px but need to find a way to adjust it when a time value is added.

**Comment by teenoh** [comment]
Thanks for the updates @MaggieAppleton! Pushed some code to handle updating the width and reducing the opacity in the scenarios you talked about. Tried applying a className (something like "input--expanded") to the input but that didn't work because react date picker updates the className when changes are made. Decided to use a data attribute to track that and style with CSS. 
Let me know if this works!


https://user-images.githubusercontent.com/16928688/172276188-61a4f24d-0d3d-4d9f-9a62-2ed0208dd128.mov

File: packages/blocks/countdown/src/base.css (hunk 2 above)
Comment (teenoh) [reply packages/blocks/countdown/src/base.css]: Deleted in [78a4bfa](https://github.com/hashintel/hash/pull/583/commits/78a4bfacddf8a51534e8c32c843440b435045144)

**Comment by MaggieAppleton** [comment]
New changes look great! Thanks for figuring out those details. Makes a big difference

**Review by kachkaev — APPROVED** [review]

**Comment by nonparibus** [comment]
@MaggieAppleton @teenoh I think the default behavior should be to hide the second-counter except when a timer is within <1hr of its target. If you are 'counting down' to the launch of a product, or a talk, or something similar, it's nice to have.

Ultimately this kind of decision/variability in function seems like something that should be made a user-specifiable preference within an EA's config pane.

We need to draw up a set of best-practice guidelines re: what to expose via config panels. cc @CiaranMn @jtewright This may even fall under our existing [design guidelines](https://hash.dev/blog/block-design), as it will invariably get into the question of "what settings should I allow users to toggle in-block" vs in the EA's config pane for the block).

**Pull request merged**